### PR TITLE
[Onnx][UnitTests] Excluded additional onnx tests

### PR DESCRIPTION
**Pull request by Lunderberg**
- The onnx tests `test_basic_convinteger`, `test_convinteger_with_padding`, `test_range_float_type_positive_delta_expanded`, and `test_range_int32_type_positive_delta_expanded` don't run correctly on CUDA targets, so they are added to the exclusion.

- Parametrized over the relative directory name, rather than the full directory name.  This improves readability of the pytest output, and keeps the same parametrized test name across different python version.

- Changed the target-specific skips to check the target kind, rather than the full target string.

### CODE REVIEW

**Comment by Lunderberg** [comment]
@mbrookhart Can you review this, especially the addition of the cuda-specific excluded tests?

File: tests/python/frontend/onnx/test_forward.py [hunk 1]
```diff
@@ -4418,317 +4419,323 @@ def verify_eyelike(indata):
...
-        for failure in target_skips[target]:
-            if failure in test:
-                pytest.skip()
-                break
-    for failure in unsupported_onnx_tests:
-        if failure in test:
-            pytest.skip()
-            break
+@pytest.mark.parametrize("onnx_test", onnx_test_folders)
+def test_onnx_nodes(target, dev, onnx_test):
```
Comment (mbrookhart) [inline tests/python/frontend/onnx/test_forward.py:4717]: I'm not seeing how target and dev get into the test?

**Review by mbrookhart — COMMENTED** [review]
One confusion on parametarization, otherwise LGTM

File: tests/python/frontend/onnx/test_forward.py (hunk 1 above)
Comment (Lunderberg) [reply tests/python/frontend/onnx/test_forward.py:4717]: This was a feature that I added in #8010 , where any unit test or fixture that accepts a `target` or `dev` argument is parametrized over all targets in `tvm.testing.enabled_targets`.  Basically, the same as the explicit `pytest.mark.parametrize` that was there previously, or the `tvm.testing.parametrize_targets`, but without needing each function to reproduce the same boilerplate.  The `target` parametrize is defined in [`tvm.testing._auto_parametrize_target`](https://github.com/apache/tvm/blob/main/python/tvm/testing.py#L788), and the `dev` fixture based on it is in the [`conftest.py`](https://github.com/apache/tvm/blob/main/conftest.py#L29).

**Comment by Lunderberg** [comment]
Closing, this clean-up will be done along-side #8542, which was based on this.

**Pull request closed without merging**